Amplify/Categories/DataStore/Model/Internal/ModelRegistry.swift

```diff
@@ -49,6 +49,21 @@ public struct ModelRegistry {
         }
     }
 
+    public static func register(modelName: ModelName,
```

Comment:
Let's refactor `register(modelType:)` to invoke this method to make sure all our "initialization" code goes through the same path.